### PR TITLE
Fixes #1055 - External Request Endpoint

### LINKED ISSUE #1055: Receive request from smoke alarm portal
@MisterJames opened https://github.com/redcross/smoke-alarm-portal/issues/196 in the "get a smoke alarm" repo, and this is the corresponding half of that work:
1.  An endpoint we can hit with each new smoke alarm request that comes in.  This endpoint would receive a request object with these fields: 
   
   ```
   name, // the name of the person requesting the alarm  
   address, // their address  
   assigned_rc_region, // the Red Cross region associated with their zip code  
   city,  
   state,  
   zip,  
   phone,  
   email,  
   serial, // our unique identifier: Red Cross region code - 2-digit fiscal year - integer  
   status // currently: new, in progress, installed, or canceled  
   ```

and respond with a status and a boolean indicating whether or not AllReady will handle this request (see [this comment](https://github.com/redcross/smoke-alarm-portal/issues/196#issuecomment-213773653) for details).
1. When the status of a request changes, you can post the new status to our `/admin/requests/status/:id` endpoint.
2. You might want to hit our endpoint at `/regions` for the list of currently active regions.

### PR DESCRIPTION
**Pull request by seanepping**
Fixes #1055 
Added RequestApiController
Added Tests for controller
Added RequestViewModel
Added RequestCommand components
Added ExternalEndpointAttribute that can be used to avoid
AntoForgeryTokenChecking in tests

### CODE REVIEW

**Comment by mheggeseth** [comment]
Hi Sean, if you add the phrase "Fixes #1055" to the description of the PR, it will automatically close the issue once the PR is merged.

**Comment by tonysurma** [comment]
Hey @seanepping you removed WIP, is this good to go?

File: AllReadyApp/Web-App/AllReady/web.config [hunk 1]
```diff
@@ -4,6 +4,6 @@
     <handlers>
       <add name="aspNetCore" path="*" verb="*" modules="AspNetCoreModule" resourceType="Unspecified" />
     </handlers>
-    <aspNetCore processPath=".\AllReady.exe" arguments="" forwardWindowsAuthToken="false" stdoutLogEnabled="false" />
+    <aspNetCore processPath="%LAUNCHER_PATH%" arguments="%LAUNCHER_ARGS%" forwardWindowsAuthToken="false" stdoutLogEnabled="false" />
```
Comment (mheggeseth) [inline AllReadyApp/Web-App/AllReady/web.config]: Please revert the changes to this file. Until we figure out the environment variable situation for our Azure deployments, we need to hardcode `processPath` and `arguments`.

File: AllReadyApp/Web-App/AllReady/ViewModels/Requests/RequestViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+namespace AllReady.ViewModels.Requests
+{
+    public class RequestViewModel
```
Comment (mheggeseth) [inline AllReadyApp/Web-App/AllReady/ViewModels/Requests/RequestViewModel.cs]: Can you add the following to this view model? If a consumer wants to provide that info, we'll take it. Otherwise, we'll use geocoding to figure it out.
```
public double Latitude { get; set; }
public double Longitude { get; set; }
```

File: AllReadyApp/Web-App/AllReady/web.config (hunk 1 above)
Comment (tonysurma) [inline AllReadyApp/Web-App/AllReady/web.config]: I just put the environment variables on azure so don't have to revert (as a test) for now.

**Comment by seanepping** [comment]
@tonysurma I believe so.  With the current setup I don't have a means to have the smoke alarm app talk to my local, but they should be able to hit a dev environment?
